src/engine/terrain.cpp

```diff
@@ -44,13 +45,19 @@ const static unsigned int chunk_indices[96] = {
     60, 62, 61, 61, 62, 63,
 };
 
-Terrain::Terrain(const std::string& pmg, const std::string& tiles) : Terrain() {
-
-}
+Terrain::Terrain(const std::string& tileset) {
+  // attempt to load in the atlas sheet
+  atlas_ = new TextureAtlas(512, 8);
+  for(unsigned int i = 0; i < 256; ++i) {
+    if(!atlas_->AddImage(tileset + std::to_string(i))) {
+      break;
+    }
+  }
+  atlas_->Finalize();
```

Comment:
Finalize function should throw a warning if there aren't any images loaded. Need to double check this!
In such a case, texture pointer needs to be our default texture.